packages/gitbook-v2/src/middleware.ts

```diff
@@ -146,20 +147,17 @@ async function serveSiteRoutes(requestURL: URL, request: NextRequest) {
 
         // We use the host/origin from the canonical URL to ensure the links are
         // correctly generated when the site is proxied. e.g. https://proxy.gitbook.com/site/siteId/...
-        const siteCanonicalURL = new URL(siteURLData.canonicalUrl);
+        const siteCanonicalURL = getSiteCanonicalURL(siteURLData, visitorToken);
 
         //
         // Make sure the URL is clean of any va token after a successful lookup
         // The token is stored in a cookie that is set on the redirect response
         //
         const incomingURL = mode === 'url' ? requestURL : siteCanonicalURL;
-        const requestURLWithoutToken = normalizeVisitorAuthURL(incomingURL);
-        if (
-            requestURLWithoutToken !== incomingURL &&
-            requestURLWithoutToken.toString() !== incomingURL.toString()
-        ) {
+        const incomingURLWithoutToken = normalizeVisitorAuthURL(incomingURL);
+        if (incomingURLWithoutToken.toString() !== incomingURL.toString()) {
```

Comment:
For proxy requests our `incoming` URL is the `siteCanonicalURL` but it does not include the va token.  So a VA site when proxied would remove the token from the URL after a successful resolution. But with this PR it fixes that